opensquirrel/circuit_builder.py

```diff
@@ -90,6 +118,10 @@ def _check_generator_f_args(generator_f: Callable[..., Gate | Measure], attr: st
                 raise TypeError(
                     f"Wrong argument type for instruction `{attr}`, got {type(args[i])} but expected {par.annotation}"
                 )
+            if args[i].__class__.__name__ == "Qubit":
+                self._check_qubit_out_of_bounds_access(args[i].index)
+            elif args[i].__class__.__name__ == "Bit":
+                self._check_bit_out_of_bounds_access(args[i].index)
```

Comment:
This code is indeed quite _ad hoc_ to the current implementation of the `CircuitBuilder`.

If I can say something, I don't really think we should spend much time with this, since we have started working on a new version of the `CircuitBuilder`. I think we should put our efforts there.

Having said that, and considering this is very much likely to change in a near future, I would be happy enough with some code that improves the current implementation although is not future proof.